metadata-models/src/main/pegasus/com/linkedin/dataset/FineGrainUpstream.pdl

```diff
@@ -0,0 +1,26 @@
+namespace com.linkedin.dataset
+
+import com.linkedin.common.AuditStamp
+import com.linkedin.common.DatasetUrn
+
+/**
+ * Upstream lineage information about a dataset including the source reporting the lineage
+ */
+record FineGrainUpstream {
```

Comment:
Let us slow down a bit and take it in steps. Let us get only the models reviewed and checked-in first.